client/engine/chainservice/eth_chainservice.go

```diff
@@ -111,25 +116,41 @@ func (ecs *EthChainService) SendTransaction(tx protocols.ChainTransaction) error
 }
 
 func (ecs *EthChainService) subcribeToEvents() error {
+
+	go ecs.listenForLogEvents()
+	return nil
+}
+
+func (ecs *EthChainService) listenForLogEvents() {
 	// Subsribe to Adjudicator events
 	query := ethereum.FilterQuery{
 		Addresses: []common.Address{ecs.naAddress},
 	}
 	logs := make(chan ethTypes.Log)
 	sub, err := ecs.chain.SubscribeFilterLogs(context.Background(), query, logs)
 	if err != nil {
-		return err
+		panic(err)
 	}
-	go ecs.listenForLogEvents(sub, logs)
-	return nil
-}
-
-func (ecs *EthChainService) listenForLogEvents(sub ethereum.Subscription, logs chan ethTypes.Log) {
 	for {
 		select {
 		case err := <-sub.Err():
-			// TODO should we try resubscribing to chain events
-			ecs.logger.Printf("event subscription error: %v", err)
+			// If the error is nil then the subscription was closed and we need to re-subscribe.
+			// This is a workaround for https://github.com/ethereum/go-ethereum/issues/23845
+			if err == nil {
```

Comment:
A level of nesting can be avoided if this is refectored to:
``` golang
if err != nil {
  panic(err)
}
//...remaining logic
```

Reply:
Updated in https://github.com/statechannels/go-nitro/pull/922/commits/0c41f84611fb6ccd6ab86b0c48530d7c02096646